pylint/lint/pylinter.py

```diff
@@ -1659,7 +1659,7 @@ def _get_messages_to_set(
         else:
             category_id_formatted = category_id
         if category_id_formatted is not None:
-            for _msgid in self.msgs_store._msgs_by_category.get(category_id_formatted):
+            for _msgid in self.msgs_store._msgs_by_category[category_id_formatted]:
```

Comment:
```suggestion
            for _msgid in self.msgs_store._msgs_by_category.get(category_id_formatted, []):
```

Reply:
It won't, as it's a defaultdict. If the key doesn't exist it'll make one by the default factory (`list()`) in this case.